vllm/model_executor/weight_utils.py

```diff
@@ -275,6 +275,107 @@ def hf_model_weights_iterator(
             torch.cuda.empty_cache()
 
 
+def kv_cache_scales_loader(
```

Comment:
This function is not simple. We should not do these validation ourselves. Please use `pydantic` library to define the schema to be read into. https://docs.pydantic.dev/latest/

Reply:
Was looking to refactor the logic with jsonschema/pydantic once a formal schema is established for quantization params so we can generalize across models and also integrate it elsewhere (potential candidates: scales extraction utility, quantization config). But good call on at least doing the basic structure checking with Pydantic for now.